src/plugin.ts

```diff
@@ -6,23 +6,41 @@ import {
 } from "prosemirror-state";
 import { getSuggestionDecorations } from "./decorations.js";
 
-export const suggestChangesKey = new PluginKey<{ enabled: boolean }>(
+export interface SuggestChangesPluginState {
+  enabled: boolean;
+  generateId: () => string;
```

Comment:
We pass our id generator (@neuledge/id tagged id generator) here

Reply:
Rather than storing this in the state, since it's only ever used in the withSuggestChanges wrapper, maybe it should just be an optional argument to that function?